### ISSUE TITLE
Time Issue with Vagrant

### ISSUE THREAD

**Issue by aaalexliu** [End-to-end-provenance/RDataTracker-vagrant#7]
So I and Moe did a quick run through of installing and opening the vagrant virtual machine, and it works in general with some confusing capture options. However, the main issue is that it took ~30 minutes to get vagrant fully up and running (i.e. from installing in brew to opening the virtual machine). The main speed bump I found was when vagrant was downloading ubuntu:

Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'boxcutter/ubuntu1604-desktop' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Loading metadata for box 'boxcutter/ubuntu1604-desktop'
    default: URL: https://atlas.hashicorp.com/boxcutter/ubuntu1604-desktop
==> default: Adding box 'boxcutter/ubuntu1604-desktop' (v2.0.18) for provider: virtualbox
    default: Downloading: https://atlas.hashicorp.com/boxcutter/boxes/ubuntu1604-desktop/versions/2.0.18/providers/virtualbox.box
==> default: Successfully added box 'boxcutter/ubuntu1604-desktop' (v2.0.18) for 'virtual box'!

The downloading part took roughly 20 minutes, but this is on fast, relatively uncrowded wifi at average speeds around 2000 Kb/sec, so I don't know how fast this would be at Dataverse. Is there a possible configuration of the virtual box/vagrant install file that comes with ubuntu pre-loaded? That would solve this issue.

Another minor problem – the OS X install file from the virtual box link doesn't verify on my computer as well as Anna's. We had to use home-brew to install virtual box.

**Comment by tfjmp** [comment]
You are downloading an operating system with a graphic GUI, there is not really any way to improve things on that front.

**Comment by erboose** [comment]
I'm also concerned that we may be sending the wrong signal to domain scientists who (a) are intimidated by such a complicated installation and/or (b) don't want to add 5+ GB of software to their personal machines.  All users really need to do to use our tools is install the eeProv package for R.

Perhaps we could offer that as an alternative using the install package file posted on the Harvard Forest website.

**Comment by tfjmp** [comment]
From yesterday meeting, I think we were to offer several alternatives.

**Comment by MKLau** [comment]
Additionally, I think that the plan to use a USB drives will avoid the lengthy install time.

Thomas, have you tested the speed that the vagrant runs at when it’s running from a USB?

On Jul 7, 2016, at 2:26 PM, Thomas Pasquier <notifications@github.com<mailto:notifications@github.com>> wrote:

From yesterday meeting, I think we were to offer several alternatives.

**Comment by tfjmp** [comment]
The vagrantfile is just a script to automate the provisioning process of VMs. As stated by @MKLau we should distribute the VM + the Virtual Box installer via USB key on the day. @MKLau I would advise them to copy the VM image to their hard-drive rather than running from the USB key.

@blernermhc @alexsiqingliu the purpose is not to distribute RDataTracker that way, but to have a controlled environment for tutorials/demos. There is many dependencies beyond the "RDataTracker" package. We need JRE 8, we need a certain version of R, certain R package installed, Margo exercise needs pdftex etc... One of the things we want to avoid is spending an hour solving every participants dependency issues/missing software etc.

If all participants were running some flavour of Linux will just have to write a script to install everything we need (which is basically half of the content of the vagrantfile), but that's unlikely to be the case. 

Someone can volunteer to write an installation manual for the tutorial, so participant can install/ensure that all required dependencies are available on their Windows/OS X/Linux machine.

**Comment by aaalexliu** [comment]
Yes all those dependencies could be an issue, I’m just wondering if there could be a way to include the GUI on the USB drive, as that was a sizable download that took me and Moe roughly 20 minutes. I was concerned that download time could impact the tutorial, though I could be wrong. I’d also be happy to write a tutorial if one wanted to just install RDataTracker, should I do it on some sort of Google Doc?

Best,
Alex Liu

**Comment by tfjmp** [comment]
To be slightly clearer, the USB file should contain an ova file [1], not the vagrant script. They will still need to install VirtualBox or use an alternative already installed on their machine. So there won't be the 20 minutes installation issue on Tuesday.

[1] https://en.wikipedia.org/wiki/Open_Virtualization_Format
